### PR TITLE
Stronger ESLint rules

### LINKED ISSUE #838: Directly returning async service methods leads to broken stack traces
**Description**
If we have a method `async foo()` and in it `return` a method `async bar()` that `throw`s an exception, `foo()` does not appear on the stack trace. Only `foo`s caller and `bar()` appear.

A more detailed explanation can be found here: https://www.even.li/posts/2020-06-04-async-stack-traces/

This problem bit me while debugging a `TypeError` in a service method as I could not find out from which controller it was called. I just got `@nestjs/core/router/router-execution-context.js` in the trace and then the service.

The only solution I could find was to always use `return await` instead of just `return` as laid out in the blog post.
I propose to turn on the [associated eslint check](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/return-await.md).

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
ESLint

### Description
This PR improves code quality by enabling more ESLint rules.
It also fixes #838 by enabling the rule mentioned there.

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)
#838

### CODE REVIEW

**Comment by davidmehren** [comment]
This PR is best reviewed commit-by-commit.

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -173,7 +175,10 @@ export class AuthService {
 
   toAuthTokenDto(authToken: AuthToken): AuthTokenDto | null {
     if (!authToken) {
-      this.logger.warn(`Recieved ${authToken} argument!`, 'toAuthTokenDto');
+      this.logger.warn(
+        `Recieved ${String(authToken)} argument!`,
```
Comment (DerMolly) [inline src/auth/auth.service.ts:179]: Shouldn't this rather be `authToken.toString()`?

File: src/groups/groups.service.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ export class GroupsService {
    */
   toGroupDto(group: Group | null | undefined): GroupInfoDto | null {
     if (!group) {
-      this.logger.warn(`Recieved ${group} argument!`, 'toGroupDto');
+      this.logger.warn(`Recieved ${String(group)} argument!`, 'toGroupDto');
```
Comment (DerMolly) [inline src/groups/groups.service.ts:48]: same as above

File: src/logger/console-logger.service.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export class ConsoleLoggerService {
     this.classContext = context;
   }
 
-  error(message: any, trace = '', functionContext?: string) {
+  error(message: unknown, trace = '', functionContext?: string) {
     this.printMessage(
       message,
       clc.red,
```
Comment (DerMolly) [inline src/logger/console-logger.service.ts:27]: `ESLint: Unsafe member access .magentaBright on an any value.(@typescript-eslint/no-unsafe-member-access)`

Maybe we should import the colors directly instead of importing clc?
`import { red } from 'cli-color';`

File: src/users/users.service.ts [hunk 4]
```diff
@@ -56,7 +56,7 @@ export class UsersService {
 
   toUserDto(user: User | null | undefined): UserInfoDto | null {
     if (!user) {
-      this.logger.warn(`Recieved ${user} argument!`, 'toUserDto');
+      this.logger.warn(`Recieved ${String(user)} argument!`, 'toUserDto');
```
Comment (DerMolly) [inline src/users/users.service.ts:59]: here again?

File: src/logger/console-logger.service.ts (hunk 3 above)
Comment (davidmehren) [reply src/logger/console-logger.service.ts:27]: That should be gone. Did you run `yarn install`?

File: src/auth/auth.service.ts (hunk 1 above)
Comment (davidmehren) [reply src/auth/auth.service.ts:179]: That does not work, as `null` and `undefined` do not have `.toString()`

File: src/logger/console-logger.service.ts (hunk 3 above)
Comment (DerMolly) [reply src/logger/console-logger.service.ts:27]: I did and this still occurs. Curiously it does not happen when I run `yarn lint`, but only as an error in the IDE.

Comment (ErikMichelson) [reply src/logger/console-logger.service.ts:27]: Maybe delete your `.eslintcache`?

Comment (DerMolly) [reply src/logger/console-logger.service.ts:27]: The backend doesn't have a `.eslintcache`or at least I don't…

File: tsconfig.json [hunk 5]
```diff
@@ -10,6 +10,6 @@
     "sourceMap": true,
     "outDir": "./dist",
     "baseUrl": "./",
-    "incremental": true
+    "incremental": true,
```
Comment (InnayTool) [inline tsconfig.json:13]: I don't think this is standards-compliant ([RFC8259](https://tools.ietf.org/html/rfc8259) page 6).

File: src/api/public/notes/notes.controller.ts [hunk 6]
```diff
@@ -188,16 +189,16 @@ export class NotesController {
...
   ): Promise<NoteMetadataDto> {
     try {
       const note = await this.noteService.getNoteByIdOrAlias(noteIdOrAlias);
       if (!this.permissionsService.mayRead(req.user, note)) {
         throw new UnauthorizedException('Reading note denied!');
       }
-      return this.noteService.toNoteMetadataDto(note);
-    } catch (e) {
+      return await this.noteService.toNoteMetadataDto(note);
+    } catch (e: unknown) {
```
Comment (InnayTool) [inline src/api/public/notes/notes.controller.ts:201]: Why is this added here? In other places there is no annotation.

File: src/main.ts [hunk 7]
```diff
@@ -49,4 +49,4 @@ async function bootstrap() {
   logger.log(`Listening on port ${appConfig.port}`, 'AppBootstrap');
 }
 
-bootstrap();
+void bootstrap();
```
Comment (InnayTool) [inline src/main.ts:52]: I think the better solution is to add the return type `Promise<void>` to the function.

**Review by InnayTool — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 7 above)
Comment (davidmehren) [reply src/main.ts:52]: That won't help here. [no-floating-promises](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md) wants us to handle all promises or explicitly ignore them with `void`.

File: tsconfig.json (hunk 5 above)
Comment (davidmehren) [reply tsconfig.json]: fixed

File: src/api/public/notes/notes.controller.ts (hunk 6 above)
Comment (davidmehren) [reply src/api/public/notes/notes.controller.ts]: removed

**Review by InnayTool — APPROVED** [review]

File: src/monitoring/monitoring.service.ts [hunk 8]
```diff
@@ -16,12 +16,20 @@ let versionCache: null | {
   preRelease?: string;
   commit?: string;
 } = null;
-async function getServerVersionFromPackageJson() {
+async function getServerVersionFromPackageJson(): Promise<{
+  major: number;
+  minor: number;
+  patch: number;
+  preRelease?: string;
+  commit?: string;
```
Comment (DerMolly) [inline src/monitoring/monitoring.service.ts:24]: Maybe this should be it's own type instead of typed out two times next to each other?

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Review by InnayTool — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]
